### PR TITLE
add setting to lower column case when reading parquet/orc file

### PR DESCRIPTION
**Pull request by shuchaome**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add setting to lower column case when reading parquet/ORC file

Generally we create table with lowercase column name, but some parquet/orc table in hive store uppercase in schema, which cause mismatch，so we need a setting to decide whether to lower column name when reading parquet/orc file.
What's more, name_to_column_ptr.begin()->second cause segment fault if all columns mismatch, better to add a check.

### CODE REVIEW

File: src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp [hunk 1]
```diff
@@ -511,19 +513,22 @@ Block ArrowColumnToCHColumn::arrowSchemaToCHHeader(const arrow::Schema & schema,
...
     NameToColumnPtr name_to_column_ptr;
-    for (const auto & column_name : table->ColumnNames())
+    for (auto column_name : table->ColumnNames())
     {
         std::shared_ptr<arrow::ChunkedArray> arrow_column = table->GetColumnByName(column_name);
         if (!arrow_column)
             throw Exception(ErrorCodes::DUPLICATE_COLUMN, "Column '{}' is duplicated", column_name);
+        if (enable_lowering_column_name)
+            boost::to_lower(column_name);
         name_to_column_ptr[column_name] = arrow_column;
```
Comment (antonio2368) [inline src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp:532]:
```suggestion
        name_to_column_ptr[std::move(column_name)] = arrow_column;
```

File: src/Processors/Formats/Impl/ORCBlockInputFormat.cpp [hunk 2]
```diff
@@ -145,7 +146,10 @@ void ORCBlockInputFormat::prepareReader()
...
         /// so we should recursively count the number of indices we need for this type.
         int indexes_count = countIndicesForType(schema->field(i)->type());
-        const auto & name = schema->field(i)->name();
+        auto name = schema->field(i)->name();
+        if (format_settings.enable_lowering_column_name)
+            boost::to_lower(name);
+
         if (getPort().getHeader().has(name) || nested_table_names.contains(name))
         {
             column_names.push_back(name);
```
Comment (antonio2368) [inline src/Processors/Formats/Impl/ORCBlockInputFormat.cpp:155]:
```suggestion
            column_names.push_back(std::move(name));
```

**Review by antonio2368 — COMMENTED** [review]
Try adding some tests for it and it's good for merge.

**Comment by shuchaome** [comment]
stateless test failed because parquet-reader wasn't built, how to fix?

> 2022-03-11 07:39:19 /usr/share/clickhouse-test/queries/0_stateless/00900_long_parquet_load.sh: line 49: /usr/share/clickhouse-test/queries/0_stateless/../../../contrib/arrow/cpp/build/release/parquet-reader: No such file or directory

**Review by antonio2368 — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Very strange. It will be more logical to make a setting to enable case insensitive column names matching. Because users may have ALL_CAPS in Parquet/ORC as well, but would you make yet another setting for this case?

`input_format_use_lowercase_column_name` the setting is named in a way so it's expected that it will work for all the formats (CSVWithNames, Native, JSONEachRow). But it is implemented only for ORC and Parquet, that is very inconsistent.

Let me remove this feature from the changelog, so the users will not have chance to accidentally start using it...

**Pull request merged**